### DISCUSSION TITLE
if not syntax

### DISCUSSION THREAD

**Discussion by hrrrrustic** [dotnet/csharplang#4143] [Language Ideas] [answered]
Since `not` keyword was introduced, it would be nice to be allowed to write like this
```C#
if not (something.IsValid)
```

instead of
```C#
if(!something.IsValid)
```

Yep, new version is a little bit longer, but it looks more readable and obvious
Also, if there are more than 1 conditions in brackets, it will be
```C#
if not (something1 op1 something2 op2 something3)
```

instead of
```C#
if (!(something1 op1 something2 op2 something3))
```

(yeah, it's possible to use de morgan's laws here, but it doesn't matter)

**Comment by canton7** [comment, marked as answer]
`if not` is basically the same as Ruby's `unless`. I'm sure there was a proposal for that, but it's really hard to search for....

The general consensus is that `unless` is more confusing than not when applied with multiple conditions (sometimes it works, but most of the time it takes some effort to parse), and is pointless when combined with `else`.

**Reply by PathogenDavid** [reply]
Yup, here's some of them: https://github.com/dotnet/csharplang/issues/157 https://github.com/dotnet/roslyn/issues/7927

**Comment by Unknown6656** [comment]
<img width="400" src="https://user-images.githubusercontent.com/8807985/99329083-c3724100-287d-11eb-8110-854f713864fc.png"/>
<sub>'twas a joke, I couldn't help it ;)</sub><br/>
<sup><i>I know it has nothing to do with the proposal</i></sup>
